### DISCUSSION TITLE
Copilot doesn't cooperate

### DISCUSSION THREAD

**Discussion by przemo098** [community/community#52862] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I'm trying to receive any code snippet or test from copilot:
![image](https://user-images.githubusercontent.com/6282239/232161000-04d25d61-ad7a-439b-b34f-75e862434543.png)

```
[DEBUG] [getCompletions] [2023-04-14T21:48:36.095Z] Requesting completion at position 10:11, between "    {\r\n        // " and "\r\n    }\r\n".
[INFO] [default] [2023-04-14T21:48:36.180Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:48:36.479Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 298 ms
[INFO] [streamChoices] [2023-04-14T21:48:36.480Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:48:36.481Z] request done: headerRequestId: [17259139-8c0a-4d4a-a8ac-882ca79753f4] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion dismissed
[DEBUG] [getCompletions] [2023-04-14T21:48:36.901Z] Requesting completion at position 10:16, between "    {\r\n        // load " and "\r\n    }\r\n".
[INFO] [default] [2023-04-14T21:48:36.989Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:48:37.241Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 250 ms
[INFO] [streamChoices] [2023-04-14T21:48:37.243Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:48:37.246Z] request done: headerRequestId: [23896d0e-8720-4778-a015-b186c79fbb06] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion rejected
[DEBUG] [getCompletions] [2023-04-14T21:48:37.601Z] Requesting completion at position 10:17, between "    {\r\n        // load f" and "\r\n    }\r\n".
[INFO] [default] [2023-04-14T21:48:37.684Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:48:37.876Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 192 ms
[INFO] [streamChoices] [2023-04-14T21:48:37.877Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:48:37.878Z] request done: headerRequestId: [83beef46-5334-40af-bc29-d414f680e002] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion dismissed
[DEBUG] [getCompletions] [2023-04-14T21:48:38.023Z] Requesting completion at position 10:21, between "    {\r\n        // load file " and "\r\n    }\r\n".
[INFO] [default] [2023-04-14T21:48:38.109Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:48:38.337Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 228 ms
[INFO] [streamChoices] [2023-04-14T21:48:38.338Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:48:38.339Z] request done: headerRequestId: [5ea33d9c-9fdc-4f08-8f00-497cc759cc82] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion accepted
[DEBUG] [agent] [2023-04-14T21:50:23.015Z] Agent service starting
[DEBUG] [agent] [2023-04-14T21:50:23.079Z] Telemetry initialized
[DEBUG] [getCompletions] [2023-04-14T21:50:23.783Z] Requesting completion at position 3:8, between "\r\n// load " and "".
[INFO] [default] [2023-04-14T21:50:24.423Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:50:24.780Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 355 ms
[INFO] [streamChoices] [2023-04-14T21:50:24.788Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:50:24.794Z] request done: headerRequestId: [75065067-f979-491a-aa85-a9f7b5e61a37] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion dismissed
[DEBUG] [getCompletions] [2023-04-14T21:50:25.475Z] Requesting completion at position 3:13, between "\r\n// load file " and "".
[INFO] [default] [2023-04-14T21:50:25.563Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-14T21:50:25.879Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 316 ms
[INFO] [streamChoices] [2023-04-14T21:50:25.881Z] solution 0 returned. finish reason: [stop]
[INFO] [streamChoices] [2023-04-14T21:50:25.882Z] request done: headerRequestId: [f36b6601-1dea-40e2-af94-56a148b8b224] model deployment ID: [x58519fc8b61f]
[INFO] [CopilotProposalSourceProvider] Completion accepted
```

any ideas ?
VS version: 17.5.4
Copilot: 1.82.0.0

**Comment by warlock8hz** [comment]
Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.5.3
GitHub Copilot 1.82.1.0 (v1.82.1.0@10c323cc0)
GitHub Copilot Agent v1.82.0

Facing exactly the same problem every time when [Tab] was pressed and returned: 
`[INFO] [CopilotProposalSourceProvider] Completion dismissed`
even happens for local inline suggestions. 

By the way, CoPilot works normally in C# environments, but does not work at all in C++. Both have Reshaper installed. 

In some projects, all shortcuts (including Ctrl+Z) are ignored after the suggestions popped out. VS indicates autoformats have been completed after pressing Tab, but nothing happens indeed.

**Reply by warlock8hz** [reply]
Nevermind, I fixed it by turning off this option after noticing that "autoformat" operation: 
Option -> Text Editor -> C/C++-> Code Style -> Formatting -> General -> Automatically indent when I type a tab 

Here is the reason (I guess): 
Every time I pressed a tab, there would be an autoformat operation, which caused CoPilot to dismiss the suggestion and changes, and the tab (the four spaces) would also be reversed since they were at the end of the line.

**Reply by robindegen** [reply]
I'd still consider that a bug though. It should play nicely with other settings regardless

**Comment by ayim** [comment, marked as answer]
We've invested a ton in quality, resolving conflicts with settings and offering more suggestions since April ... I'm going to close this issue for now.

**Reply by robindegen** [reply]
There is still an issue where _something_ is cancelling my copilot suggestions half the time. It was introduced relatively recently and its really annoying

**Reply by ayim** [reply]
What language are ya writing in?

**Reply by robindegen** [reply]
C++; in Visual Studio 2022. It doesn't always happen; but it happens occasionally. I see a completion for a split second and it disappears; and then the log says it was cancelled. If i try again by just doing backspace/space sometimes it sticks

**Reply by ayim** [reply]
And you're on VS 17.8?

**Reply by robindegen** [reply]
17.8.5

**Reply by robindegen** [reply]
In my coding session just now it kept happening again:

```
[DEBUG] [getCompletions] [2024-02-07T00:37:16.046Z] Requesting completion at position 16:9, between "        m_data.Int64[2] = 0;\r\n        m" and "\r\n    }\r\n".
[INFO] [default] [2024-02-07T00:37:16.136Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2024-02-07T00:37:16.329Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 193 ms
[INFO] [streamChoices] [2024-02-07T00:37:16.330Z] solution 0 returned. finish reason: [stop]
[INFO] [CopilotProposalSourceProvider] Completion dismissed
[INFO] [streamChoices] [2024-02-07T00:37:16.330Z] request done: headerRequestId: [070bc91e-6c0f-409e-9e39-73b473be6d94] model deployment ID: [x5aa6d170bcae]
```

I think it is clashing with something in Resharper C++; but i'm not really sure what exactly.

**Reply by ayim** [reply]
Mmmm..... are you working on anything where you might be able to record a screenshare?
If you need a tool, PowerPoint's Media > Insert > Screen Recording would be my go to.

We'll want to see this twice: 
Iteration 1: In Visual Studio, click View > Output > GitHub Copilot
Iteration 2: View > Output > Intellode

**Reply by robindegen** [reply]
The intellicode outputs nothing it seems. But I added it for completion. In both gifs you should see the line pop up for a split second and then copilot logs it was dismissed. I did not press any buttons. If i type a bit more sometimes it does pick it up.

(edit: i realised the framerate wasn't good enough. I will rerecord; one moment)

**Reply by robindegen** [reply]
![dismissed_code](https://github.com/community/community/assets/5236908/445ef217-492d-4610-88be-096d252bc53b)

**Reply by robindegen** [reply]
Intellicode outputs nothing:
![dismissed_code_intellicode](https://github.com/community/community/assets/5236908/70522364-097d-4811-b7ae-e024b128636c)

**Reply by ayim** [reply]
Amazing videos - opened a ticket here since it repros.
https://developercommunity.visualstudio.com/t/C-GitHub-Copilot-Completion-appears-qu/10580132?

Gotta ask, does this occur even if you disable resharper?

**Reply by robindegen** [reply]
I'm not fully sure yet. The thing is that its not 100% reproducable. When it does happen its really bad and it happens all the time. Last night it wasn't happening at all. The time I recorded those vids turning off resharper did help from what I could tell; but I'd have to keep trying. I created a keybind for suspending and resuming resharper so I can try a bit more.
